gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/api/use_case/UpdateDynamicPropertiesUseCaseTest.java

```diff
@@ -0,0 +1,279 @@
+    private ApiStateDomainService apiStateDomainService = new ApiStateDomainServiceLegacyWrapper(apiStateService);
+
+    private UpdateDynamicPropertiesUseCase cut;
+
+    @BeforeEach
+    void setUp() {
+        apiStateService = mock(ApiStateService.class);
+        apiStateDomainService = new ApiStateDomainServiceLegacyWrapper(apiStateService);
+        environmentCrudServiceInMemory.initWith(
+            List.of(Environment.builder().id("environment-id").organizationId("organization-id").build())
+        );
+        cut =
+            new UpdateDynamicPropertiesUseCase(
+                apiCrudServiceInMemory,
+                apiStateDomainService,
+                environmentCrudServiceInMemory,
+                new AuditDomainService(auditCrudServiceInMemory, userCrudServiceInMemory, new JacksonJsonDiffProcessor()),
+                apiEventQueryServiceInMemory
+            );
+    }
+
+    @AfterEach
+    void tearDown() {
+        Stream
+            .of(
+                apiCrudServiceInMemory,
+                environmentCrudServiceInMemory,
+                auditCrudServiceInMemory,
+                userCrudServiceInMemory,
+                apiEventQueryServiceInMemory
+            )
+            .forEach(InMemoryAlternative::reset);
+    }
+
+    @Test
+    void should_not_update_api_if_same_properties() {
+        final Api api = buildApiWithProperties(List.of(Property.builder().key("key").value("value").dynamic(true).build()));
+        apiCrudServiceInMemory.initWith(List.of(api));
+        cut.execute(
+            new UpdateDynamicPropertiesUseCase.Input(
+                api.getId(),
+                HTTP_DYNAMIC_PROPERTIES,
+                List.of(Property.builder().key("key").value("value").dynamic(true).build())
+            )
+        );
+
+        assertThat(auditCrudServiceInMemory.storage()).isEmpty();
+        verifyNoInteractions(apiStateService);
+    }
+
+    @Test
+    void should_update_api_with_new_properties_but_not_deploy_because_synchronized() {
+        final Api api = buildApiWithProperties(List.of(Property.builder().key("user-prop").value("value").dynamic(false).build()));
+        apiCrudServiceInMemory.initWith(List.of(api));
+        when(apiStateService.isSynchronized(any(), any())).thenReturn(true);
+        cut.execute(
+            new UpdateDynamicPropertiesUseCase.Input(
+                api.getId(),
+                HTTP_DYNAMIC_PROPERTIES,
+                List.of(Property.builder().key("key").value("value").dynamic(true).build())
+            )
+        );
+
+        assertThat(apiCrudServiceInMemory.get(api.getId()).getApiDefinitionV4().getProperties())
+            .containsExactlyInAnyOrder(
+                Property.builder().key("user-prop").value("value").dynamic(false).build(),
+                Property.builder().key("key").value("value").dynamic(true).build()
+            );
+        assertAuditHasBeenCreated();
+        verify(apiStateService).isSynchronized(any(), any());
```

Comment:
No need to verify the call, it is "polluting" the test, if it was not called, the test would have failed because it would have return false